core/src/test/java/apoc/trigger/TriggerTest.java

```diff
@@ -129,40 +129,59 @@ public void testTimeStampTrigger() throws Exception {
         });
     }
 
-    // TODO NC:Why is this failing?
-    @Ignore
     @Test
-    public void testTxId() throws Exception {
-        db.executeTransactionally("CALL apoc.trigger.add('txinfo','UNWIND $createdNodes AS n SET n.txId = $transactionId, n.txTime = $commitTime',{phase:'after'})");
+    public void testTxIdAfterAsync() throws Exception {
+        db.executeTransactionally("CALL apoc.trigger.add('txinfo','UNWIND $createdNodes AS n SET n.txId = $transactionId, n.txTime = $commitTime',{phase:'afterAsync'})");
         db.executeTransactionally("CREATE (f:Bar)");
-        TestUtil.testCall(db, "MATCH (f:Bar) RETURN f", (row) -> {
+        org.neo4j.test.assertion.Assert.assertEventually(() ->
+                        db.executeTransactionally("MATCH (n:Bar) RETURN n", Map.of(),
+                                result -> {
+                                    final Node node = result.<Node>columnAs("n").next();
+                                    return (long) node.getProperty("txId", -1L) > -1L
+                                            && (long) node.getProperty("txTime") > start;
+                                })
+                , (value) -> value, 30L, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testTxId() {
+        db.executeTransactionally("CREATE (f:Another)");
+        db.executeTransactionally("CALL apoc.trigger.add('txinfo','UNWIND $createdNodes AS n \n" +
+                "MATCH (a:Another) WITH a, n\n" +
+                "SET a.txId = $transactionId, a.txTime = $commitTime',{phase:'after'})");
+        db.executeTransactionally("CREATE (f:Bar)");
+        TestUtil.testCall(db, "MATCH (f:Another) RETURN f", (row) -> {
             assertEquals(true, (Long) ((Node) row.get("f")).getProperty("txId") > -1L);
             assertEquals(true, (Long) ((Node) row.get("f")).getProperty("txTime") > start);
         });
+        db.executeTransactionally("MATCH (n:Another) DELETE n");
     }
 
     @Test
     public void testMetaDataBefore() {
-        testMetaData("before");
+        db.executeTransactionally("CALL apoc.trigger.add('txinfo','UNWIND $createdNodes AS n SET n.label = labels(n)[0], n += $metaData', {phase: 'before'})");
+        testMetaData("MATCH (n:Bar) RETURN n");
     }
 
-    // TODO NC:Why is this failing?
-    @Ignore
     @Test
     public void testMetaDataAfter() {
-        testMetaData("after");
+        db.executeTransactionally("CREATE (n:Another)");
+        db.executeTransactionally("CALL apoc.trigger.add('txinfo', 'UNWIND $createdNodes AS n MATCH (a:Another) SET a.label = labels(n)[0], a += $metaData', {phase: 'after'})");
```

Comment:
How come we can modify the node here adding things if it is phase after?

Reply:
As above, because the node `(a:Another)` is not present in `$createdNodes`, but is created before the triggered query.